src/engine/board.ts

```diff
@@ -5,10 +5,11 @@ import Piece from './pieces/piece';
 
 export default class Board {
     public currentPlayer: Player;
+    public lastMovedPiece: Piece | null = null
     private readonly board: (Piece | undefined)[][];
 
-    public constructor() {
-        this.currentPlayer = Player.WHITE;
+    public constructor(currentPlayer?: Player) {
+        this.currentPlayer = currentPlayer ? currentPlayer : Player.WHITE;
```

Comment:
I don't think you wrote this but just FYI as it's useful this can be simplified to `currentPlayer || Player.WHITE`
`X || Y`  means if X then X else Y